torch/nn/parallel/distributed.py

```diff
@@ -396,7 +402,12 @@ def __init__(
         ).format(distinct_device_types)
         self.device_type = list(distinct_device_types)[0]
 
-        if self.device_type == "cpu" or self.is_multi_device_module:
+        if (
+            device_ids is None
+            or len(device_ids) == 0  # For backward compatibility.
```

Comment:
I see. As long as we still preserves BC, this change to `None` LGTM. But, let's leave at least one test that uses an empty list as `device_ids` to ensure test coverage.

Reply:
I did create a new small unit test `test_nccl_backend_single_device_module_empty_device_ids` in test_c10d.py to cover this empty list case.